packages/commerce-sdk-react/src/hooks/ShopperBaskets/mutation.test.ts

```diff
@@ -1,267 +1,224 @@
-                                mutation
-                            }
-                        })
-
-                        await waitForValueToChange(() => result.current.queries.basket.data)
-
-                        act(() => {
-                            result.current.mutation.mutate(payload)
-                        })
-
-                        await waitForValueToChange(() => result.current.mutation.isSuccess)
-                        expect(result.current.mutation.isSuccess).toBe(true)
-                        // On successful mutation, the query cache gets updated too. Let's assert it.
-                        const cacheUpdateMatrix = getCacheUpdateMatrix(CUSTOMER_ID)
-                        // @ts-ignore
-                        const matrixElement = cacheUpdateMatrix[mutationName](payload, {})
-                        const {invalidate, update, remove}: CacheUpdateMatrixElement = matrixElement
-
-                        const assertionData = {
-                            basket: newBasket,
-                            customerBaskets: newCustomerBaskets
-                        }
-                        update?.forEach(({name}) => {
-                            // @ts-ignore
-                            assertUpdateQuery(result.current.queries[name], assertionData[name])
-                        })
-
-                        invalidate?.forEach(({name}) => {
-                            // @ts-ignore
-                            assertInvalidateQuery(result.current.queries[name], oldCustomerBaskets)
-                        })
-
-                        remove?.forEach(({name}) => {
-                            // @ts-ignore
-                            assertRemoveQuery(result.current.queries[name])
-                        })
-                    }
-                },
-                {
-                    name: 'error',
-                    assertions: async () => {
-                        mockMutationEndpoints('/checkout/shopper-baskets/', {errorResponse: 500})
-
-                        const {result, waitForNextUpdate} = renderHookWithProviders(() => {
-                            const action = mutationName as ShopperBasketsMutationType
-                            return useShopperBasketsMutation({action})
-                        })
-
-                        act(() => {
-                            result.current.mutate(payload)
-                        })
-
-                        await waitForNextUpdate()
-
-                        expect(result.current.error).toBeDefined()
-                    }
-                }
-            ]
-        }
-    }
-)
-
-tests.forEach(({hook, cases}) => {
-    describe(hook, () => {
-        beforeEach(() => {
-            jest.clearAllMocks()
-        })
-        cases.forEach(({name, assertions}) => {
-            test(name, assertions)
-        })
+const deleteTestCase = ['deleteBasket', makeOptions<'deleteBasket'>(undefined, {})] as const
```

Comment:
Aren't `remove...` where we call delete to the api counted as deleteTestCase? What makes this one different then the `removeItemFromBasket?

Reply:
`removeItemFromBasket` and the other endpoints _modify_ the basket (by removing nested data), and they all return `Basket`. Only `deleteBasket` _deletes_ the basket, and you're correct, it returns `void` (since there's no `Basket` to return).